### PR TITLE
The basic version of the data-api-image

### PR DESCRIPTION
**Pull request by mik-dass**
Signed-off-by: mik-dass mrinald7@gmail.com

### CODE REVIEW

File: cico_run_eslint.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+set -ex
+
+npm run check_lint
```
Comment (anmolbabu) [inline cico_run_eslint.sh]: you might need to do npm install first to pull in eslint as a dependency

File: main.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+var express = require("express"),
+	app     = express()
+
+app.use(express.json())
+app.use(express.urlencoded({ extended: true }))
+
+var port = process.env.HDD_SUPERVISOR_PORT || 9090,
```
Comment (anmolbabu) [inline main.js]: change the port env var name to anything non- supervisor and hdd

File: main.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+	ip   = process.env.IP   || process.env.OPENSHIFT_NODEJS_IP || "0.0.0.0"
+
+var routes = require("./node_modules/data-api/src/api/routes")(app)
```
Comment (anmolbabu) [inline main.js]: ./node_modules/data-api/src/api/routes suffices or js extension(to complete full file path) is required?

File: package.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  "description": "Instantiation of data-api along with its respective build and deploy scripts reside here",
+  "dependencies": {
+    "express": "^4.16.2",
+    "validator": "^9.2.0",
+    "hypothesis-core-image": "github:fabric8-hdd/hypothesis-core-image",
+    "data-api": "github:fabric8-hdd/data-api"
+  },
+  "scripts": {
+    "start": "node main.js",
+    "test": "mocha --timeout 5000 tests/*_test.js"
```
Comment (anmolbabu) [inline package.json]: latest versions of mocha don't return the control after running tests.
So you might have to append  --exit at the end like in https://github.com/fabric8-hdd/manager-api-image/blob/master/package.json#L8

File: tests/api_test.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
+var server = require("../main"),
+    chai = require("chai"),
+    chaiHTTP = require("chai-http"),
+    should = chai.should()
+
+chai.use(chaiHTTP)
+
+describe("Test data api", function () {
+
+    describe("GET to /api/v1.0/hypothesis/22f82a83-7b1b-4556-aeea-5b3b16a9b287", () => {
```
Comment (anmolbabu) [inline tests/api_test.js:10]: replace hypothesis with probably data or hypothesis-results in api path

**Comment by anmolbabu** [comment]
Rest looks good to me
But verified?

File: main.js (hunk 3 above)
Comment (mik-dass) [reply main.js]: Not required. Works for me.

File: tests/api_test.js (hunk 5 above)
Comment (mik-dass) [reply tests/api_test.js:10]: I will have to change the endpoints in data-api

**Comment by mik-dass** [comment]
@anmolbabu Yes verified and images are pushed, now fixing the ones mentioned above

**Review by anmolbabu — APPROVED** [review]
LGTM

**Pull request merged**